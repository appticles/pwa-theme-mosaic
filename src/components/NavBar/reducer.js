import { MENU_OPEN, MENU_CLOSE } from './action';

export default function (state = null, action) {
  switch (action.type) {
  case MENU_OPEN:
    return true;
  case MENU_CLOSE:
    return false;
  default:
    return state;
  }
}
