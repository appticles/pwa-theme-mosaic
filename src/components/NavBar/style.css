.navbar.button.tiny.right.ui {
  position: absolute;
  z-index: 1;
  top: 2%;
  right: 3%;
  padding-top: 10px;
  padding-bottom: 10px;
  padding-right: 16px;
  padding-left: 16px; }
  .navbar.button.tiny.right.ui .icon.content.large {
    margin: 0; }
