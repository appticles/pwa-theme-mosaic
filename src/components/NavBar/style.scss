@import "../../utilities/_variables.scss";

.navbar.button.tiny.right.ui {
  position: absolute;
  z-index: 1;
  top: 2%;
  right: 3%;
  padding-top: $base-spacing-xsm;
  padding-bottom: $base-spacing-xsm;
  padding-right: $base-spacing;
  padding-left: $base-spacing;
  .icon.content.large {
    margin: 0;
  }
}
