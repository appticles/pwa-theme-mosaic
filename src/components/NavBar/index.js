import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Button, Icon } from 'semantic-ui-react';

import { openMenu } from './action';
import './style.css';

class NavBar extends Component {
  constructor(props) {
    super(props);
    this.showSidebar = this.showSidebar.bind(this);
  }

  showSidebar(e) {
    e.stopPropagation();
    this.props.openMenu();
  }

  render() {
    return (
      <Button floated="right" size="tiny" className="navbar">
        <Icon name="content" onClick={this.showSidebar} size="large" />
      </Button>
    );
  }
}

NavBar.propTypes = {
  openMenu: PropTypes.func.isRequired,
};

export default connect(null, { openMenu })(NavBar);
