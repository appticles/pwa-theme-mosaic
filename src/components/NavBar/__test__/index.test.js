import React from 'react';
import renderer from 'react-test-renderer';

import mockStore from '../../../configureMockStore';

import NavBar from '../index';

describe('Navbar component', () => {
  it('should render correctly', () => {
    const store = mockStore({
      navbar: true,
    });

    expect(renderer.create(<NavBar store={store} openMenu={jest.fn()} />)).toMatchSnapshot();
  });
});
