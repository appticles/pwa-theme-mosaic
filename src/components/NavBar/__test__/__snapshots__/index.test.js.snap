// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component should render correctly 1`] = `
<button
  className="ui teal tiny right floated button navbar"
  disabled={undefined}
  onClick={[Function]}
  role="button"
  tabIndex={undefined}
>
  <i
    aria-hidden="true"
    className="content large icon"
    onClick={[Function]}
  />
</button>
`;
